AlphaWallet/TokenScriptClient/Coordinators/EventSourceCoordinatorForActivities.swift

```diff
@@ -139,18 +139,11 @@ extension EventSourceCoordinatorForActivities.functional {
                     return when(resolved: promises).map(on: queue, { values -> [EventActivityInstance] in
                         values.compactMap { $0.optionalValue }.compactMap { $0 }
                     })
-                }).then(on: queue, { events -> Promise<Void> in
-                    if events.isEmpty {
-                        return .value(())
-                    } else {
-                        return eventsDataStore.add(events: events, forTokenContract: tokenContract).then(on: queue, { _ -> Promise<Void> in
-                            return .value(())
-                        })
-                    }
-                }).done { _ in
+                }).done(on: .main, { events in
+                    eventsDataStore.add(events: events, forTokenContract: tokenContract)
```

Comment:
The semantics is slightly different from the original. The original returns a promise and only when that is resolved, this promise resolves (with the `return .value(())`). The replacement doesn't. Does the caller depend on this behavior? eg. insert into database and display it right away?